pkg/edgeview/src/system.go

```diff
@@ -938,6 +967,217 @@ func dispAFile(f os.FileInfo) {
 	fmt.Printf("%s, %v, %d, %s\n", f.Mode().String(), f.ModTime(), f.Size(), f.Name())
 }
 
+func getTarFile(opt string) {
+	execCmd := strings.SplitN(opt, "tar/", 2)
+	if len(execCmd) != 2 {
+		fmt.Printf("tar needs a / and path input\n")
+		return
+	}
+	source := execCmd[1]
+	printColor(" - tar cmd: "+source, colorCYAN)
+	closePipe(true)
+
+	fi, err := os.Stat(source)
+	if err != nil {
+		fmt.Printf("find directory error: %v\n", err)
+		return
+	}
+
+	if !fi.IsDir() {
+		fmt.Printf("the path %s is not a directory\n", source)
+		return
+	}
+
+	ok := checkBlockedDirs(source)
+	if !ok {
+		fmt.Printf("the directory %s can not be tarred\n", source)
+		return
+	}
+
+	// check the directory size not to exceed the max
+	sizeBytes := du(source, fi)
+	if sizeBytes > tarMaxBytes {
+		fmt.Printf("directory size %d, exceeds maximum %d\n", sizeBytes, tarMaxBytes)
+		return
+	}
+	baseDir := filepath.Base(source)
+	targetDir := "/tmp/"
+	dirs := strings.Split(source, "/")
+	tarfileName := targetDir
+	for _, d := range dirs {
+		if d == "" {
+			continue
+		}
+		tarfileName = tarfileName + d + "."
+	}
+	tarfileName = tarfileName + strconv.FormatInt(time.Now().Unix(), 10) + ".tar"
+	tarFile, err := os.Create(tarfileName)
+	if err != nil {
+		log.Errorf("can not create tar file")
+		return
+	}
+	defer tarFile.Close()
+
+	tarball := tar.NewWriter(tarFile)
+	defer tarball.Close()
+
+	err = filepath.Walk(source,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			// skip file with suffix of '.key.pem' or '.key'
+			ok := checkBlockedFileSuffix(info.Name())
+			if !ok {
+				return nil
+			}
+
+			// some files like socket type, can not be tarred
+			if info.Size() == 0 && !info.Mode().IsRegular() {
+				return nil
+			}
+			header, err := tar.FileInfoHeader(info, info.Name())
+			if err != nil {
+				return err
+			}
+
+			header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, source))
+			if err := tarball.WriteHeader(header); err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				// skip if it is part of the blocked directories
+				if !checkBlockedDirs(filepath.Join(path, info.Name())) {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+
+			// skip symlink files
+			if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+				return nil
+			}
+
+			file, err := os.Open(path)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+
+			_, err = io.Copy(tarball, file)
+			return err
+		})
+
+	tarFile.Close()
+	tarball.Close()
+	if err != nil {
+		_ = os.Remove(tarfileName)
+		log.Errorf("tar oper error: %v", err)
+		return
+	}
+
+	fi, err = os.Stat(tarfileName)
+	if err == nil {
+		fmt.Printf(" tar file size %d, compressing...\n", fi.Size())
+		closePipe(true)
+	}
+	// compress it into gzip file
+	gzFileName, err := gzipFile(tarfileName, targetDir)
+	if err != nil {
+		_ = os.Remove(tarfileName)
+		log.Errorf("gzip oper error: %v", err)
+		return
+	}
```

Comment:
Should we remove the `gzFileName` in case of error during compression?
In general, it would be great to have some kind of cleaner task or at least cleanup of the dir with files on start (if you will move it from tmp as I recommended in another comment).